### PR TITLE
Fix extended loop_control with includes

### LINKED ISSUE #61218: ansible_loop undefined with extended loop_control in an include
##### SUMMARY
Setting `extended: yes` in `loop_control` doesn't provide `ansible_loop` inside the loop.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
loop_control

##### ANSIBLE VERSION
```
ansible 2.9.0.dev0
  config file = None
  configured module search path = ['/home/kustodian/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/kustodian/git/github/ansible/lib/ansible
  executable location = /home/kustodian/git/github/ansible/bin/ansible
  python version = 3.7.4 (default, Jul 16 2019, 07:12:58) [GCC 9.1.0]
```
This also doesn't work on 2.8.4

##### CONFIGURATION
```
```

##### OS / ENVIRONMENT
ArchLinux

##### STEPS TO REPRODUCE
Create two files:
```yaml
# play.yml
---
- hosts: localhost
  connection: local
  tasks:
    - include_tasks: inner.yml
      loop:
        - 1
        - 2
        - 3
      loop_control:
        extended: yes
```
```
# inner.yml
- debug:
    var: ansible_loop
```
and run:
```
ansible-playbook play.yml -i localhost,
```

##### EXPECTED RESULTS
Ansible should print the value of `ansible_loop` three times.

##### ACTUAL RESULTS
It actually prints that `ansible_loop` is undefined:
```
$ ansible-playbook play.yml -i localhost,

PLAY [localhost] *******************************************************************************************************************************************************************************************

TASK [Gathering Facts] *************************************************************************************************************************************************************************************
ok: [localhost]

TASK [include_tasks] ***************************************************************************************************************************************************************************************
included: /tmp/inner.yml for localhost
included: /tmp/inner.yml for localhost
included: /tmp/inner.yml for localhost

TASK [debug] ***********************************************************************************************************************************************************************************************
ok: [localhost] => {
    "ansible_loop": "VARIABLE IS NOT DEFINED!"
}

TASK [debug] ***********************************************************************************************************************************************************************************************
ok: [localhost] => {
    "ansible_loop": "VARIABLE IS NOT DEFINED!"
}

TASK [debug] ***********************************************************************************************************************************************************************************************
ok: [localhost] => {
    "ansible_loop": "VARIABLE IS NOT DEFINED!"
}

PLAY RECAP *************************************************************************************************************************************************************************************************
localhost                  : ok=7    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
```

### PR DESCRIPTION
**Pull request by kustodian**
##### SUMMARY
Fixes https://github.com/ansible/ansible/issues/61218

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
loop_control

### CODE REVIEW

File: test/integration/targets/loop_control/inner.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+- debug:
+    msg: "{{ ansible_loop.index }} {{ ansible_loop.index0 }} {{ ansible_loop.revindex }} {{ ansible_loop.revindex0 }} {{ ansible_loop.first }} {{ ansible_loop.last }} {{ ansible_loop.length }} {{ ansible_loop.allitems|join(',') }}"
```
Comment (bcoca) [inline test/integration/targets/loop_control/inner.yml]: i would change to asserts, but this seems fine also

Comment (sivel) [reply test/integration/targets/loop_control/inner.yml]: Probably only need to assert that `ansible_loop is defined`

File: test/integration/targets/loop_control/runme.sh [hunk 2]
```diff
@@ -9,3 +9,14 @@ MATCH='foo_label
...
+#ok: [localhost] => {
+#    "msg": "1 0 3 2 True False 3 first,second,third"
+#ok: [localhost] => {
+#    "msg": "2 1 2 1 False False 3 first,second,third"
+#ok: [localhost] => {
+#    "msg": "3 2 1 0 False True 3 first,second,third"
+MATCH='"1 0 3 2 True False 3 first,second,third"
+"2 1 2 1 False False 3 first,second,third"
+"3 2 1 0 False True 3 first,second,third"'
+[ "$(ansible-playbook extended.yml "$@" |grep '"msg":'|sed -e 's/^.*"msg": \(.*\)$/\1/')" == "${MATCH}" ]
```
Comment (sivel) [inline test/integration/targets/loop_control/runme.sh]: These types of tests are always hard to diagnose failures with.  I'd rather this exist in the playbook itself.  The `ansible_loop` tests, use assert on all of the values, so I'm not sure we need to assert that each are still the same.

**Comment by kustodian** [comment]
I can change it to asserts, not a problem.

File: test/integration/targets/loop_control/inner.yml (hunk 1 above)
Comment (kustodian) [reply test/integration/targets/loop_control/inner.yml]: I added tests with `assert`. I tried to add as many tests I could think of without complicating things.

File: test/integration/targets/loop_control/extended.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+- name: loop_contro/extended/include https://github.com/ansible/ansible/issues/61218
```
Comment (sivel) [inline test/integration/targets/loop_control/extended.yml]:
```suggestion
- name: loop_control/extended/include https://github.com/ansible/ansible/issues/61218
```

File: test/integration/targets/loop_control/extended.yml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  hosts: localhost
+  gather_facts: false
+  tasks:
+    - name: loop on an include
+      include: inner.yml
```
Comment (sivel) [inline test/integration/targets/loop_control/extended.yml]:
```suggestion
      include_tasks: inner.yml
```

**Review by sivel — CHANGES_REQUESTED** [review]

File: test/integration/targets/loop_control/extended.yml (hunk 4 above)
Comment (kustodian) [reply test/integration/targets/loop_control/extended.yml]: Sorry, I have no idea how I forgot to use `include_tasks`. Update and pushed. If the tests pass, this should be good to go.

**Review by sivel — APPROVED** [review]

**Pull request merged**